### ISSUE TITLE
Crash when turning on In-Flight trajectory display while in Map view

### ISSUE THREAD

**Issue by fat-lobyte** [neuoy/KSPTrajectories#102]
KSP Version:  1.3.0.1804 x64
Trajectories Version: 1.6.8
Mods: None

Steps to reproduce:

1. Launch a new Vessel
2. Get to Orbit or to a Suborbital Trajectory
3. Open Map Vew
4. Enable In-Flight trajectory

Result:
Game Crashes

**Comment by fat-lobyte** [comment]
Something's fucky with the newly created FlightOverlay code. Specifically, here:
https://github.com/neuoy/KSPTrajectories/blob/master/Plugin/FlightOverlay.cs#L101

The crash occurs *after* `FixedUpdate()` has completed. My wild guess is that it's something about the Vertices Array that makes the Line object access invalid locations.
I'm not actually sure that I'll "fix" this issue. I'll probably just make sure to not execute this code when we're in MapView, I hope this works around it.

Additionally, I'm still planning to replace the in-flight display with [MechJebs GLUtils drawing functions](https://github.com/MuMech/MechJeb2/blob/master/MechJeb2/GLUtils.cs#L125) which on first glance look like a wonderful fit for us.

**Comment by PiezPiedPy** [comment]
I've noticed the crash only happens when out of the atmosphere,
 so maybe some objects/data related to when the craft is in the atmosphere gets deleted when it leaves the atmosphere, making any accesses to now non existent areas cause a crash.

**Comment by PiezPiedPy** [comment]
So....  While in Map view, out of the atmosphere, Trajectories not previously activated during current scene and In-Flight is enabled.
While `line.enable` is true any changes to `vertices` or `line.SetVertexCount(vertices.Length);` or `line.SetPositions(vertices);` will cause the crash.
If `line.enable` is first set to false before any changes to `vertices` or `line` then the crash only happens at the next `line.enable = true;` instruction.

I then tried moving the `line.enable = true;` to an `Update` method which also crashed and then tried `OnGUI` which funny enough kind of works until you move the map camera view >.<

I decided on using the work around you suggested and all seems to be fine. I'll do a PR soon
